src/ProjectNeon/Assets/Features/Effects/Attack.cs

```diff
@@ -1,18 +1,42 @@
-﻿using System.Collections;
-using System.Collections.Generic;
+﻿
+
+
 using UnityEngine;
-using UnityEngine.Events;
 
-public sealed class Attack 
+public sealed class Attack  : Effect
 {
-    public Member Attacker { get; }
-    public Member Target { get; }
+    private Member _attacker;
+    public Member Attacker() { return this._attacker; }
+    private Target _target;
+    public Target Target() { return this._target; }
     public int Damage { get; }
 
-    public Attack(Member attacker, Member target, int damage)
+    public Attack(int damage)
     {
-        Attacker = attacker;
-        Target = target;
         Damage = damage;
     }
+
+    public void Apply(Member source, Target target)
```

Comment:
This is a very complex method and seems to have at least 3 jobs.

Can we have a design that follows the Single Responsibility Principle?

Reply:
@SilasReinagel `Attack.Apply` behavior is complex anyway: we have to trigger `Attack`  for each `Member` we have in `Target`, deal with the  `Attack`  and notify `BattleEvent` about it.

Unfortunately the design forces me to make this sweep on `Target.Member`. Applying an `Effect` should use polimorphism so each `Target` would know how to apply the effects on itself, making the `Attack.Apply` method simpler and cleaner. This change would probably solve the issues pointed in https://github.com/EnigmaDragons/ProjectNeon/pull/458/files#r342374573 , https://github.com/EnigmaDragons/ProjectNeon/pull/458/files#r342374381, 
https://github.com/EnigmaDragons/ProjectNeon/pull/458/files#r342374655 and
https://github.com/EnigmaDragons/ProjectNeon/pull/458/files#r342374676 
, but I think that it gets out of the scope of the issue, which is to implement `Attack` as an `Effect`

Reply:
This isn't a big problem. We can fix it later.